packages/stream-api/src/streamApi.ts

```diff
@@ -36,80 +50,131 @@ export class StreamApi {
     return this.onResponseDispatcher.on(handler);
   }
 
+  public startStream(data: StartStreamPayload): Promise<void>;
+
   public startStream(
-    data: vertexvis.protobuf.stream.IStartStreamPayload
-  ): Promise<void> {
+    data: StartStreamPayload,
+    withResponse: boolean
+  ): Promise<vertexvis.protobuf.stream.IStreamResponse>;
+
+  public startStream(...args: unknown[]): unknown {
+    const request = this.parseArgs<
+      vertexvis.protobuf.stream.StartStreamPayload
+    >(args);
     return this.sendRequest({
-      startStream: {
-        ...data,
-      },
+      requestId: request.requestId,
+      startStream: { ...request.data },
     });
   }
 
-  public beginInteraction(): Promise<void> {
+  public beginInteraction(): Promise<void>;
+
+  public beginInteraction(
+    withResponse: boolean
+  ): Promise<vertexvis.protobuf.stream.IStreamResponse>;
+
+  public beginInteraction(...args: unknown[]): unknown {
+    const request = this.parseArgs<
+      vertexvis.protobuf.stream.BeginInteractionPayload
+    >(args);
     return this.sendRequest({
-      beginInteraction: {},
+      requestId: request.requestId,
+      beginInteraction: { ...request.data },
     });
   }
 
-  public replaceCamera({
-    camera,
-  }: vertexvis.protobuf.stream.IUpdateCameraPayload): Promise<void> {
-    return this.sendRequest({
-      updateCamera: {
-        camera,
-      },
-    });
+  public replaceCamera({ camera }: ReplaceCameraPayload): Promise<void>;
+
+  public replaceCamera(
+    { camera }: ReplaceCameraPayload,
+    withResponse: boolean
+  ): Promise<vertexvis.protobuf.stream.IStreamResponse>;
+
+  public replaceCamera(...args: unknown[]): unknown {
+    const request = this.parseArgs<
+      vertexvis.protobuf.stream.IUpdateCameraPayload
+    >(args);
+    const requestPayload = {
+      requestId: request.requestId,
+      updateCamera: { ...request.data },
+    };
+    return this.sendRequest(requestPayload);
```

Comment:
It appears we can simplify this implementation quite a bit now :) I don't think there's a need to use method overloading anymore if we use a default for the `withResponse` argument.

IMO we should default `withResponse` to `true`. I think the majority of situations you're going to want a response, and ignoring the response is a more exceptional use-case.

```ts
  public replaceCamera(
    { camera }: ReplaceCameraPayload,
    withResponse: boolean = true
  ): unknown {
    return this.sendRequest({ updateCamera: camera }, withResponse);
  }

  private sendRequest(
    request: vertexvis.protobuf.stream.IStreamRequest,
    withResponse: boolean
  ): Promise<void | vertexvis.protobuf.stream.IStreamResponse> {
    if (withResponse) {
      return new Promise(resolve => {
        const requestId = UUID.create();
        const subscription = this.onResponse(response => {
          if (request.requestId?.value == requestId) {
            resolve(response);
            subscription.dispose();
          }
        });
        this.websocket.send(
          vertexvis.protobuf.stream.StreamMessage.encode({ request }).finish()
        );
      });
    } else {
      this.websocket.send(
        vertexvis.protobuf.stream.StreamMessage.encode({ request }).finish()
      );
      return Promise.resolve();
    }
  }
```

Reply:
yeah, I was thinking that way, just not all the way there yet :)